ft-sdk/src/session.rs

```diff
@@ -0,0 +1,119 @@
+pub use ft_sys_shared::TRACKER_KEY;
+
+#[derive(Clone, Debug)]
+pub struct SessionID(pub String);
+
+impl SessionID {
+    /// Create a new session entry with the given user ID.
+    /// If the user ID is None, the session will be created without a user ID.
+    pub fn new(
+        conn: &mut ft_sdk::Connection,
+        user_id: Option<ft_sdk::auth::UserId>,
+        data: Option<serde_json::Value>,
+    ) -> Result<Self, ft_sdk::Error> {
+        use diesel::prelude::*;
+        use ft_sdk::schema::fastn_session;
+
+        let session_id = ft_sdk::utils::uuid_v8();
+
+        let data = match data {
+            Some(d) => serde_json::to_string(&d)?,
+            None => "{}".to_string(),
+        };
+
+        let user_id = user_id.map(|u| u.0);
+
+        diesel::insert_into(fastn_session::table)
+            .values((
+                fastn_session::id.eq(&session_id),
+                fastn_session::uid.eq(user_id),
+                fastn_session::created_at.eq(ft_sdk::env::now()),
+                fastn_session::updated_at.eq(ft_sdk::env::now()),
+                fastn_session::data.eq(data),
+            ))
+            .execute(conn)?;
+
+        Ok(Self(session_id))
+    }
+
+    pub fn from_string<S: AsRef<str>>(s: S) -> Self {
+        Self(s.as_ref().to_string())
+    }
+
+    /// Set the user ID for the given session.
+    pub fn set_user_id(
+        &self,
+        conn: &mut ft_sdk::Connection,
+        user_id: ft_sdk::auth::UserId,
+    ) -> Result<SessionID, ft_sdk::Error> {
+        self.update_one(conn, Some(user_id), None)
+    }
+
+    /// Set the data for the given session.
+    pub fn set_data(
```

Comment:
`set_data()` and `data()` are a bit low level. We usually want to look up our own key in the session, not get the entire session data. Add `get_key()` and `set_key()` also. I am even thinking should we only allow `get_key()` and `set_key()`, so blast radius of buggy code can be reduced (they can corrupt their own key, where as a bug around `set_data()` corrupts entire session data.

Reply:
I agree, we should have `get_key()` and `set_key()` instead of this :+1:

Reply:
these have been added in https://github.com/FifthTry/ft-sdk/pull/52/commits/25f7f56d9c8d472115d553d782b5296e5aaeef03. I've kept `SessionID::data` as that may still be useful to avoid multiple db calls.